src/quacc/schemas/ase.py

```diff
@@ -270,7 +271,10 @@ def summarize_opt_run(
     """
 
     additional_fields = additional_fields or {}
-    opt_parameters = dyn.todict() | {"fmax": dyn.fmax}
+    if isinstance(dyn, Optimizer):
+        dyn_parameters = dyn.todict() | {"fmax": dyn.fmax}
+    if isinstance(dyn, MolecularDynamics):
```

Comment:
Perhaps this line would be better if it is an `else` so that we are more general.